### DISCUSSION TITLE
DOC: about documentation

### DISCUSSION THREAD

**Discussion by ylme** [ejoy/ant#19] [General]
ant 引擎目前对于新手游戏开发者应该是有一定的上手门槛的。所以这里先不聊如何通过 documentation 降低新手开发者的上手门槛（可以另开一个 discussion 去聊），而是先假定使用的开发者有一定的游戏开发经验，然后怎样通过 documentation 帮助开发者了解 ant 引擎提供的功能，去做点事情。

我个人有如下一些建议。

### 使用引擎可执行文件

* ant 引擎提供了哪些可执行文件，项目组可以怎样集成这些可执行文件，并生成自己的游戏可执行文件（是否和打包相关）？
* 使用 ant 引擎时，策划，美术，程序的工作流是怎样的？如何使用编辑器？
* 如何导入美术资源和使用？
* UI 的工作流是怎样的？

### 使用引擎编程接口

* 既然开源的，大家必然会对实现感兴趣，引擎的总体架构以及设计理念是怎样的？
* ant 引擎的特色，特别是在手机上面直接看效果这块，开发者如何去使用，以及总体实现是怎样的？
* 引擎基于 ECS 这块，可以从总体上介绍一下 ECS 在引擎内部的应用。
* **开发者可以调用哪些 API ，应该会分成好几个模块，比如图像渲染 API ，资源管理 API ，修改游戏数据的 API （比如修改坐标），以及输入接口 API （键盘鼠标）等**？这里参考的就是通过 skynet.lua 和 socket.lua 就可以去了解可以使用的接口。这块有没有可能通过类似 doxygen 方式，导出一份接口文档出来呢？
* UI 模块的总体实现？以及一些 UI 框架实践示例？
* 资源管理和加载的总体实现？
* 渲染的总体实现？

以上是个人的一些建议。

**Comment by ylme** [comment]
还有一点，有没有可能说引擎内置一些 primitive 。参考 [Unity-CreatePrimitive](https://docs.unity3d.com/ScriptReference/GameObject.CreatePrimitive.html) 和 [Unity-PrimitiveObject](https://docs.unity3d.com/Manual/PrimitiveObjects.html) 。

比如就有只有一个 cube 时，就可以拼出来一个场景出来。然后把 cube 当作一个 player ，把 sphere 当作发射的子弹，就可以编写一个小玩具游戏了。

**Reply by cloudwu** [reply]
Ant 本身有一套 [VFS](https://github.com/ejoy/ant/wiki/VFS) 机制 ，引擎本身有一个包叫 ant.resources.binary ，里面有一些预制件，包括了 cube 这些东西。https://github.com/ejoy/ant.resources/tree/master/meshes/base

你写的 demo 只要引用了引擎这个包在它自己的 vfs 上，就可以直接使用它们了。

**Comment by cloudwu** [comment]
我正在以类似思路写文章：可以打开仓库 wiki 页签查看。限于写作速度，很难一蹴而就，慢慢来吧。

原来也是计划年后发布的，所以也可以理解为只是代码仓库提前公开了。

**Reply by cloudwu** [reply]
目前，以上绝大部分内容在 https://github.com/ejoy/ant/wiki 中都已经写了。如有遗漏，可以在 discussion 中提问，再在 wiki 中补充。
